docs/using-hive.md

```diff
@@ -2,162 +2,326 @@
-  1. Create a Kubernetes secret containing your AWS credentials:
-        ```bash
-        oc create secret generic mycluster-aws-creds --from-literal=aws_secret_access_key=$AWS_SECRET_ACCESS_KEY --from-literal=aws_access_key_id=$AWS_ACCESS_KEY_ID
-        ```
-     **NOTE**: This will be done automatically if using `hiveutil create-cluster`.
-  1. Create a [PersistentVolume](https://kubernetes.io/docs/concepts/storage/persistent-volumes/) for your `ClusterDeployment` to store the installation logs. The `accessModes` is `ReadWriteOnce` for your `PersistentVolume`. Note that if you do not want to capture must-gather logs then you can set `.spec.failedProvisionConfig.skipGatherLogs` to `true` in the `HiveConfig`.
-### Using Global Pull Secret
-
-GlobalPullSecret can be used to specify a pull secret that will be used globally by all of the cluster deployments created by Hive.
-When GlobalPullSecret is defined in Hive namespace and a cluster deployment specific pull secret is specified, the registry authentication in both secrets will be merged and used by the new OpenShift cluster.
-When a registry exists in both pull secrets, precedence will be given to the contents of the cluster specific pull secret.
-
-The global pull secret must live in the "hive" namespace, to create one:
+Cluster provisioning begins when a caller creates a `ClusterDeployment` CR, which is the core Hive resource used to control the lifecycle of a cluster and the Hive API entrypoint.
+
+Hive comes with an optional `hiveutil` binary to assist creating the `ClusterDeployment` and its dependencies. See the [hiveutil documentation](hiveutil.md) for more information.
+
+### DNS
+
+OpenShift installation requires a live and functioning DNS zone in the cloud account into which you will be installing the new cluster(s). For example if you own example.com, you could create a hive.example.com subdomain in Route53, and ensure that you have made the appropriate NS entries under example.com to delegate to the Route53 zone. When creating a new cluster, the installer will make future DNS entries under hive.example.com as needed for the cluster(s).
+
+In addition to the default OpenShift DNS support, Hive offers a DNS feature called Managed DNS. With Managed DNS, Hive can automatically create delegated zones for approved base domains. For example, if hive.example.com exists and is specified as your managed domain, you can specify a base domain of cluster1.hive.example.com on your `ClusterDeployment`, and Hive will create this zone for you, add forwarding records in the base domain, wait for it to resolve, and then proceed with installation.
+
+### Pull Secret
+
+OpenShift installation requires a pull secret obtained from try.openshift.com. You can specify an individual pull secret for each cluster Hive creates, or you can use a global pull secret that will be used by all of the clusters Hive creates.
```

Comment:
"can be used by any", rather than "will be used by all".

Reply:
Actually I might be wrong here, if you setup a global, it will be used by all, ignore this please!